pkg/test/case.go

```diff
@@ -33,16 +34,27 @@ type Case struct {
 	Dir        string
 	SkipDelete bool
 	Timeout    int
+	Namespace  string
 
 	Client          func(forceNew bool) (client.Client, error)
 	DiscoveryClient func() (discovery.DiscoveryInterface, error)
 
 	Logger testutils.Logger
 }
 
+type namespace struct {
+	Name        string
+	AutoCreated bool
```

Comment:
The code is clearer now, I am still concerned that `Case struct` contains a `Namespace` field, and there is also an `ns` variable. The distinction between their purpose is not clear at first sight.

If you don't think we could eliminate one of these, maybe we could at least rename `Case.Namespace` to something line `Case.NamespaceSpec` to make it clearer that it is a recipe for discovering/generating the actual namespace to use?

Reply:
it isn't a namespace "spec"... but I get your point... the issue to be considered is if we make `case.namespace` a `namespace` type, than do we make `TestSuite` of that type... I'm real hesitate at this point, because this is externally exposed in the test suite file.  Let me think on this one.

Reply:
hmm... after more thought I think I understand why your suggestion better on naming... I'm real hesitate to use the "spec" word as it carries different meaning in the k8s world.  still thinking...